core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -703,6 +705,35 @@ public TableMetadata removeSnapshotLogEntries(Set<Long> snapshotIds) {
         snapshots, newSnapshotLog, addPreviousFile(file, lastUpdatedMillis));
   }
 
+  private PartitionSpec reassignPartitionIds(PartitionSpec partitionSpec, AtomicInteger lastPartitionId) {
```

Comment:
Minor: this is private, but I think it would be a bit better to use `TypeUtil.NextID` just because that removes confusion about whether `incrementAndGet` should be used or `getAndIncrement`.

Reply:
Updated.